public/js/action-list.js

```diff
@@ -28,194 +28,622 @@
+                if (! list) {
+                    return;
+                }
+
+                let activeItems = list.querySelectorAll('[data-action-item].active');
+
+                if (
+                    ! activeItems.length
+                    || (_this.matchesDetailUrl(list.dataset.icingaDetailUrl, detailUrl.path)
+                        && detailUrl.query !== '?' + activeItems[0].dataset.icingaDetailFilter
+                    )
+                    || (
+                        list.dataset.icingaMultiselectUrl === detailUrl.path
+                        && detailUrl.query !== _this.createMultiSelectUrl(activeItems, false)
+                    )
+                ) {
+                    _this.clearSelection(activeItems);
+                } else {
+                    return;
+                }
+            } else {
+                list = container.querySelector('.action-list');
+            }
 
             if (list && list.matches('[data-icinga-multiselect-url], [data-icinga-detail-url]')) {
-                let _this = event.data.self;
-                let detailUrl = _this.icinga.utils.parseUrl(_this.icinga.history.getCol2State().replace(/^#!/, ''));
+                let detailUrl = _this.icinga.utils.parseUrl(
+                    _this.icinga.history.getCol2State().replace(/^#!/, '')
+                );
 
+                let item = null;
                 if (list.dataset.icingaMultiselectUrl === detailUrl.path) {
                     for (const filter of parseSelectionQuery(detailUrl.query.slice(1))) {
-                        let item = list.querySelector('[data-icinga-multiselect-filter="' + filter + '"]');
+                        item = list.querySelector('[data-icinga-multiselect-filter="' + filter + '"]');
                         if (item) {
                             item.classList.add('active');
                         }
                     }
                 } else if (_this.matchesDetailUrl(list.dataset.icingaDetailUrl, detailUrl.path)) {
-                    let item = list.querySelector('[data-icinga-detail-filter="' + detailUrl.query.slice(1) + '"]');
+                    item = list.querySelector('[data-icinga-detail-filter="' + detailUrl.query.slice(1) + '"]');
                     if (item) {
                         item.classList.add('active');
                     }
                 }
-            }
 
-            if (list && isTopLevelContainer && list.matches('[data-icinga-multiselect-url]')) {
-                let activeItems = list.querySelectorAll('[data-action-item].active');
-
-                if (activeItems.length) {
-                    if (!container.querySelector('.footer')) {
-                        container.appendChild(notjQuery.render(
-                            '<div class="footer" data-action-list-automatically-added></div>'
-                        ));
-                    }
+                if (item) {
+                    _this.scrollItemIntoView(item);
```

Comment:
Not sure if this is a good idea. The content could be rendered based on an autorefresh, and then the scroll position must be preserved, and not changed randomly. (The item may move it's position in the list, depending on the sort by rules)